articles/130_ros_time.md

```diff
@@ -115,23 +127,38 @@ It is possible that the user may have access to an out of band time source which
 It might be possible that for their use case a more advanced algorithm would be needed to propagate the simulated time with adequate precision or latency with restricted bandwidth or connectivity.
 The user will be able to switch out the time source for either each instance of their Time object as well as have the ability to override the default for the process.
 
-It is possible to use an external time source such as GPS in as a ROSTime source, but it is recommended to integrate a time source like that using standard ntp integrations with the system clock since that is already an established mechanism and will not need to deal with more complicated changes such as time jumps.
+It is possible to use an external time source such as GPS in as a ROSTime source, but it is recommended to integrate a time source like that using standard NTP integrations with the system clock since that is already an established mechanism and will not need to deal with more complicated changes such as time jumps.
+
+For the current implementation a `TimeSource` API will be defined such that it can be overridden in code.
+If in the future a common implementation is found that would be generally useful it could be extended to optionally dynamically select the alternative TimeSource via a parameter similar to enabling the simulated time.
 
 ## Implementation
 
 The `SystemTime`, `SteadyTime`, and `ROSTime` API's will be provided by each client library in an idiomatic way, but they may share a common implementation, e.g. provided by `rcl`.
 However, if a client library chooses to not use the shared implementation then it must implement the functionality itself.
 
+`SteadyTime` will be typed differently than the interchangable `SystemTime` and `ROSTime`.
```

Comment:
Can you elaborate how these types will be different. Do they share a common base in order to use them interchangeably? Is the API each of them offers different?

Reply:
This was from discussions with @wjwwood. It makes sense to have SystemTime and ROSTime have a common base class with runtime checks that they are correctly comparable. However SteadyTime is never comparable to SystemTime or SteadyTime so it would actually have a separate implementation of the same API. Thus you could get protection from misusing them at compile time instead of only catching it at runtime. 

There could also potentially be some ways to implement this same protection using template logic for the C++ API.

Reply:
The rational makes sense to me. Can you please add this to the article.

Reply:
+1 for adding to the article, at least that they should not be compared and by having separate types we avoid that at compile time.